libtransact/src/context/manager/thread/mod.rs

```diff
@@ -272,3 +272,163 @@ impl ContextManager {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::collections::HashMap;
+
+    use crate::protocol::receipt::{Event, EventBuilder, StateChange};
+    use crate::state::hashmap::HashMapState;
+
+    static KEY1: &str = "111111111111111111111111111111111111111111111111111111111111111111";
+    static KEY2: &str = "222222222222222222222222222222222222222222222222222222222222222222";
+    static KEY3: &str = "333333333333333333333333333333333333333333333333333333333333333333";
+    static KEY4: &str = "444444444444444444444444444444444444444444444444444444444444444444";
+
+    static BYTES1: [u8; 4] = [0x01, 0x02, 0x03, 0x04];
+    static BYTES2: [u8; 4] = [0x05, 0x06, 0x07, 0x08];
+    static BYTES3: [u8; 4] = [0x09, 0x10, 0x11, 0x12];
+
+    static EVENT_TYPE1: &str = "sawtooth/block-commit";
+    static ATTR1: (&str, &str) = (
+        "block_id",
+        "f40b90d06b4a9074af2ab09e0187223da7466be75ec0f472 \
+         f2edd5f22960d76e402e6c07c90b7816374891d698310dd25d9b88dce7dbcba8219d9f7c9cae1861",
+    );
+    static ATTR2: (&str, &str) = ("block_num", "3");
+
+    fn make_event() -> Event {
+        EventBuilder::new()
+            .with_event_type(EVENT_TYPE1.to_string())
+            .with_attributes(vec![
+                (ATTR1.0.to_string(), ATTR1.1.to_string()),
+                (ATTR2.0.to_string(), ATTR2.1.to_string()),
+            ])
+            .with_data(BYTES1.to_vec())
+            .build()
+            .unwrap()
+    }
+
+    fn check_state_change(state_change: StateChange) {
+        match state_change {
+            StateChange::Set { key, value } => {
+                assert_eq!(KEY2, key);
+                assert_eq!(BYTES2.to_vec(), value);
+            }
+            StateChange::Delete { key } => {
+                assert_eq!(KEY1, key);
+            }
+        }
+    }
+
+    fn check_transaction_receipt(transaction_receipt: TransactionReceipt, event: Event) {
+        for state_change in transaction_receipt.state_changes {
+            check_state_change(state_change)
+        }
+        assert_eq!(vec!(event), transaction_receipt.events);
+        assert_eq!(vec!(BYTES3.to_vec()), transaction_receipt.data);
+    }
+
+    /// Tests the functionality of the threaded Context Manager implemented through
+    /// the ContextLifecycle trait. The Context Manager's join handle is then used to shut down the
+    /// Context Manager thread.
+    #[test]
+    fn test_context_lifecycle() {
+        let state = HashMapState::new();
+        let (join_handle, context_manager) = ContextManagerJoinHandle::new(Box::new(state));
+
+        let state_id = HashMapState::state_id(&HashMap::new());
+
+        let context_id = context_manager.create_context(&[], &state_id);
+        assert!(context_id.is_ok());
```

Comment:
Rather than calling `assert!(_.is_ok())`, you can just do `_.expect("failed to ...")`